lemur/manage.py

```diff
@@ -188,216 +187,308 @@ def generate_settings():
+            if self.save_other_options:
+                # grab everything up to the next option
+                while state.rargs and not done:
+                    for prefix in self._eat_all_parser.prefixes:
+                        if state.rargs[0].startswith(prefix):
+                            done = True
+                    if not done:
+                        value.append(state.rargs.pop(0))
+            else:
+                # grab everything remaining
+                value += state.rargs
+                state.rargs[:] = []
+            value = tuple(value)
+
+            # call the actual process
+            self._previous_parser_process(value, state)
+
+        retval = super(OptionEatAll, self).add_to_parser(parser, ctx)
+        for name in self.opts:
+            our_parser = parser._long_opt.get(
+                name) or parser._short_opt.get(name)
+            if our_parser:
+                self._eat_all_parser = our_parser
+                self._previous_parser_process = our_parser.process
+                our_parser.process = parser_process
+                break
+        return retval
+
+
+def options_from_db(options):
+    map_to_types = dict(
+        array=str,
+        number=float,
+        string=str,
     )
 
-    def run(self, name, users, description):
-        user_objs = []
-        for u in users:
-            user_obj = user_service.get_by_username(u)
-            if user_obj:
-                user_objs.append(user_obj)
-            else:
-                sys.stderr.write("[!] Cannot find user {0}".format(u))
-                sys.exit(1)
-        role_service.create(name, description=description, users=users)
-        sys.stdout.write("[+] Created new role: {0}".format(name))
+    def decorator(f):
+        for opt_params in reversed(options):
+            param_decls = (
+                '-' + opt_params['short'],
+                '--' + opt_params['long'],
+                opt_params['name'])
+            attrs = dict(
+                required=opt_params['required'],
+                type=map_to_types.get(
+                    opt_params['type'], opt_params['type'])
+            )
+            if opt_params['type'] == 'array':
+                attrs['cls'] = OptionEatAll
+                attrs['nargs'] = -1
+
+            click.option(*param_decls, **attrs)(f)
+        return f
+
+    return decorator
+
+
+run_options = [
+    {
+        "name": "thename1",
+        "short": "a",
+        "long": "ace",
+        "type": "string",
+        "required": False
+    }, {
+        "name": "thename2",
+        "short": "b",
+        "long": "bravo",
+        "type": "number",
+        "required": True
+    }, {
+        "name": "thename3",
+        "short": "c",
+        "long": "candy",
+        "type": "array",
+        "required": True
+    }
+]
+
+
+def options_from_class():
+    def decorator(f):
+        click.option("--test", required=True, type=str)(f)
+        return f
+    return decorator
```

Comment:
Are these intentional inclusions?

Reply:
sorry this was test code, forgot to remove 🤦